patternfly/patternfly-elements
# feat: Optimize and simplify the build scripts

### What has changed and why

Summarize files edited as part of this MR along with a brief description of what was changed/why.

* Updates the demo and live-demo scripts to accept individual components as input

### Testing instructions

1. `npm run demo`
    - Compiles all elements
    - Spins up the demo server, triggers http://localhost:8000/examples
    - Spins up the storybook instance, triggers http://localhost:9001/?path=/story/*
    - Try making updates to any element, no recompile should be triggered (watch is not active)
2. `npm run demo pfe-cta`
    - Compiles all pfe-cta ONLY
    - See test 1
3. `npm run demo pfe-cta pfe-card`
    - Compiles all pfe-cta and pfe-card ONLY
    - See test 1
4. `npm run live-demo`
    - Compiles all elements
    - Spins up the demo server, triggers http://localhost:8000/examples
    - Spins up the storybook instance, triggers http://localhost:9001/?path=/story/*
    - Try making updates to any element, a recompile should be triggered (watch is active)
5. `npm run live-demo pfe-cta`
    - Compiles all pfe-cta ONLY
    - See test 4
6. `npm run live-demo pfe-cta pfe-card`
    - Compiles all pfe-cta and pfe-card ONLY
    - See test 4

#### Regression testing

1. `npm run dev`
    - Should compile all elements
    - Should spin up the demo server, triggers http://localhost:8000/examples
2. `npm run dev pfe-cta`
    - Should compile ONLY the pfe-cta
    - Should spin up the demo server, triggers http://localhost:8000/examples
3. `npm run start`
    - Should spin up the demo server, triggers http://localhost:8000/examples
4. `npm run build`
    - Should compile all elements
5. `npm run build pfe-cta`
    - Should compile ONLY the pfe-cta

#### Infrastructure requirements

Your repository infrastructure updates should work for at least:

### Ready-for-merge Checklist

Check off items as they are completed.  Feel free to delete items if they are not applicable to your PR.


**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
